beprof/profile.py

```diff
@@ -144,11 +146,14 @@ def normalize(self, dt):
         except RuntimeWarning as e:
             logger.error('in normalize(). self class is {0}, dt={1}'.format(self.__class__, dt))
             raise Exception("Scaling factor error:\n" + str(e))
-        self.y /= ave
+        self.y = self.y / ave
```

Comment:
Ok, I have to ideas for `rescale` and `normalize` divisions:
```py
    def rescale(self, factor=1.0, allow_cast=True):
        if allow_cast:  # will work in 99.9% cases
            try:
                self.y /= factor
            # there is this case, where "TypeError: ufunc 'true_divide' output (typecode 'd') could not be coerced
            # to provided output parameter (typecode 'l') according to the casting rule ''same_kind''" i thrown
            except TypeError as e: 
                logger.info("Division in place is impossible, casting...\n%s" % e)
                self.y = self.y / factor
        elif not np.issubdtype(self.y.dtype.type, type(factor)):
            raise TypeError("Type mismatch - array dtype: %s, factor type: %s\nallow_cast flag set to True should help"
                            % (self.y.dtype.type, type(factor)))
        else:
            try:
                self.y /= factor
            except TypeError as e:
                raise TypeError("%s\nallow_cast flag set to True should help" % e)
```
and simpler(?) one:
```py
def rescale(self, factor=1.0, allow_cast=True):
    try:
        self.y /= factor
    except TypeError as e: 
        logger.info("Division in place is impossible.\n%s" % e)
        if allow_cast:
            self.y = self.y / factor
        else:
            raise TypeError("%s\nallow_cast flag set to True should help" % e)
```